src/frontend/components/Chat/SharedChatComponents/ChatConversationDisplayer/index.tsx

```diff
@@ -1,27 +1,39 @@
 import { Box, Spinner } from 'grommet';
 import React, { useEffect, useRef, useState } from 'react';
 
-import { ChatMessageItem } from 'components/Chat/SharedChatComponents/ChatMessageItem';
-import { useMessagesState } from 'data/stores/useMessagesStore';
+import { ChatMessageGroupItem } from 'components/Chat/SharedChatComponents/ChatMessageGroupItem';
+import { ChatPresenceItem } from 'components/Chat/SharedChatComponents/ChatPresenceItem';
+import { chatItemType, useChatItemState } from 'data/stores/useChatItemsStore';
+import { HISTORY_MESSAGES_TIMEOUT_IN_MS } from 'default/chat';
+import { report } from 'utils/errors/report';
 
 export const ChatConversationDisplayer = () => {
-  const { messages } = useMessagesState();
+  const { chatItems, hasReceivedMessageHistory } = useChatItemState();
   const [loading, setLoading] = useState(true);
   const scrollableContainerRef = useRef<HTMLDivElement>(null);
 
+  // When whole message history is received, it adds presences to the chat
   useEffect(() => {
-    setTimeout(() => {
+    if (hasReceivedMessageHistory) {
       setLoading(false);
-      scrollableContainerRef.current!.scrollTo({
-        top: scrollableContainerRef.current!.scrollHeight,
-      });
-    }, 3000);
+    }
+  }, [hasReceivedMessageHistory]);
+
+  // If the message history is not received, an error message should be displayed
+  useEffect(() => {
+    setTimeout(() => {
+      if (!useChatItemState.getState().hasReceivedMessageHistory) {
+        setLoading(false);
+        report('Unable to retrieve message history.');
+        // And prompt an error modal
+      }
+    }, HISTORY_MESSAGES_TIMEOUT_IN_MS);
   }, []);
 
   const isScrollInBottom = scrollableContainerRef.current
     ? scrollableContainerRef.current!.scrollHeight -
-        Math.round(scrollableContainerRef.current!.scrollTop) ===
-      scrollableContainerRef.current!.clientHeight
+        Math.round(scrollableContainerRef.current!.scrollTop) <=
+      scrollableContainerRef.current!.clientHeight + 10
```

Comment:
I removed exclamation marks (they are useless here, because this just what the ternary is all about), but i stayed with the ternary. Indeed, when the component is first render the ref is null, so assertion needs to be done.